components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -1343,4 +1343,32 @@ private int readHighestPriorityOfPolicies() throws PolicyManagerDAOException {
         return priority;
     }
 
+    @Override
+    public int getPolicyCount() throws PolicyManagerDAOException {
+
+        Connection conn;
+        PreparedStatement stmt = null;
+        ResultSet resultSet = null;
+        int policyCount = 0;
+        try {
+            conn = this.getConnection();
+            String query = "SELECT COUNT(DM_POLICY.ID) FROM DM_POLICY";
+            stmt = conn.prepareStatement(query);
+            resultSet = stmt.executeQuery();
+
+            while (resultSet.next()) {
+                policyCount = resultSet.getInt(0);
+            }
+            return policyCount;
+
+        } catch (SQLException e) {
+            String msg = "Error occurred while reading the policies from the database.";
+            log.error(msg, e);
+            throw new PolicyManagerDAOException(msg, e);
+        } finally {
+            PolicyManagementDAOUtil.cleanupResources(stmt, resultSet);
+            this.closeConnection();
```

Comment:
How is it possible for us to have a method like this that doesn't take a reference of a "Connection" as a parameter, in. What that means is, a global reference to a connection is maintained somewhere within this particular class which is not a good approach. You might have to either extend "cleanupResources" method to support connection closing by sending the relevant Connection references in, or have a separate method to do the same.